File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/paper/entity/PaperQuestion.java
package com.yyjy.exam.entity.paper.entity;

import com.yyjy.exam.entity.question.entity.Questions;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;

@Entity
public interface PaperQuestion {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int id();
	
	@OnDissociate(DissociateAction.DELETE)
	@ManyToOne
	Paper paper();
	
	@ManyToOne
	@OnDissociate(DissociateAction.DELETE)
	Questions question();
	
	double score();
	
	@Nullable
	LocalDateTime createTime();
	
	@Nullable
	LocalDateTime updateTime();
	
	@Column(name = "deleted_at")
	@LogicalDeleted("now")
	@Nullable
	LocalDateTime delFlag();
}

File: exam-system-server/entity/src/main/kotlin/com/yyjy/models/entity/PaperQuestion.kt
package com.yyjy.models.entity

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * 试卷题目关联表（试卷包含的题目）
 */
@Entity
interface PaperQuestion {

    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    val id: Int

    @ManyToOne
    val paper: com.yyjy.models.entity.Paper

    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    val question: com.yyjy.models.entity.Questions

    val score: Double

    /**
     * 创建时间
     */
    val createTime: LocalDateTime?

    /**
     * 更新时间
     */
    val updateTime: LocalDateTime?

    /**
     * 删除时间
     */
    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    val delFlag: LocalDateTime?
}

